tests/test_wave_03.py

```diff
@@ -0,0 +1,782 @@
+
+
+def test_get_second_page_of_customers(client, one_customer, second_customer):
+    # Arrange
+    data = {"count": 1, "page_num": 2}
+
+    # Act
+    response = client.get("/customers", query_string = data)
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert len(response_body) == 1
+    assert response_body[0]["name"] == CUSTOMER_2_NAME
+    assert response_body[0]["id"] == CUSTOMER_2_ID
+    assert response_body[0]["phone"] == CUSTOMER_2_PHONE
+    assert response_body[0]["postal_code"] == CUSTOMER_2_POSTAL_CODE
+
+    
+def test_get_first_page_of_customers_grouped_by_two(client, one_customer, second_customer, third_customer):
+    # Arrange
+    data = {"count": 2, "page_num": 1}
+
+    # Act
+    response = client.get("/customers", query_string = data)
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert len(response_body) == 2
+    assert response_body[0]["name"] == CUSTOMER_1_NAME
+    assert response_body[0]["id"] == CUSTOMER_1_ID
+    assert response_body[0]["phone"] == CUSTOMER_1_PHONE
+    assert response_body[0]["postal_code"] == CUSTOMER_1_POSTAL_CODE
+
+    assert response_body[1]["name"] == CUSTOMER_2_NAME
+    assert response_body[1]["id"] == CUSTOMER_2_ID
+    assert response_body[1]["phone"] == CUSTOMER_2_PHONE
+    assert response_body[1]["postal_code"] == CUSTOMER_2_POSTAL_CODE
+
+def test_get_second_page_of_customers_grouped_by_two(client, one_customer, second_customer, third_customer):
+    # Arrange
+    data = {"count": 2, "page_num": 2}
+
+    # Act
+    response = client.get("/customers", query_string = data)
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert len(response_body) == 1
+    assert response_body[0]["name"] == CUSTOMER_3_NAME
+    assert response_body[0]["id"] == CUSTOMER_3_ID
+    assert response_body[0]["phone"] == CUSTOMER_3_PHONE
+    assert response_body[0]["postal_code"] == CUSTOMER_3_POSTAL_CODE
+
+
+def test_get_customers_no_page(client, one_customer, second_customer, third_customer):
+    # Arrange
+    data = {"count": 2}
+
+    # Act
+    response = client.get("/customers", query_string = data)
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert len(response_body) == 2
+    assert response_body[0]["name"] == CUSTOMER_1_NAME
+    assert response_body[0]["id"] == CUSTOMER_1_ID
+    assert response_body[0]["phone"] == CUSTOMER_1_PHONE
+    assert response_body[0]["postal_code"] == CUSTOMER_1_POSTAL_CODE
+
+    assert response_body[1]["name"] == CUSTOMER_2_NAME
+    assert response_body[1]["id"] == CUSTOMER_2_ID
+    assert response_body[1]["phone"] == CUSTOMER_2_PHONE
+    assert response_body[1]["postal_code"] == CUSTOMER_2_POSTAL_CODE
+
+def test_get_customers_sorted_and_paginated(client, one_customer, second_customer, third_customer):
+    # Arrange
+<<<<<<< HEAD
```

Comment:
Looks like a missed merge conflict here. I believe we want the query param with count.